atlasdb-coordination-impl/src/main/java/com/palantir/atlasdb/coordination/keyvalue/KeyValueServiceCoordinationStore.java

```diff
@@ -0,0 +1,201 @@
+ * This {@link SequenceAndBound} should contain the ID of the value that is considered to be the most current
+ * in the sequence.
+ */
+// TODO (jkong): Coordination stores should be able to clean up old values.
+public final class KeyValueServiceCoordinationStore implements CoordinationStore {
+    private static final Logger log = LoggerFactory.getLogger(KeyValueServiceCoordinationStore.class);
+
+    private static final TableMetadata COORDINATION_TABLE_METADATA = new TableMetadata(
+            NameMetadataDescription.create(ImmutableList.of(
+                    new NameComponentDescription.Builder()
+                            .componentName("sequence")
+                            .type(ValueType.BLOB)
+                            .logSafety(TableMetadataPersistence.LogSafety.SAFE)
+                            .build())),
+            new ColumnMetadataDescription(new DynamicColumnDescription(
+                    NameMetadataDescription.create(
+                            ImmutableList.of(
+                                    new NameComponentDescription.Builder()
+                                            .componentName("sequenceNumber")
+                                            .logSafety(TableMetadataPersistence.LogSafety.SAFE)
+                                            .type(ValueType.VAR_LONG)
+                                            .build())),
+                    ColumnValueDescription.forType(ValueType.VAR_LONG))),
+            ConflictHandler.IGNORE_ALL,
+            TableMetadataPersistence.CachePriority.WARM,
+            false,
+            0,
+            false,
+            TableMetadataPersistence.SweepStrategy.NOTHING, // we do our own cleanup
+            false,
+            TableMetadataPersistence.LogSafety.SAFE);
+    private static final long WRITE_TIMESTAMP = 0L;
+    private static final ObjectMapper OBJECT_MAPPER = ObjectMappers.newServerObjectMapper();
+
+    private final KeyValueService kvs;
+    private final byte[] coordinationSequence;
+
+    private KeyValueServiceCoordinationStore(KeyValueService kvs, byte[] coordinationSequence) {
+        this.kvs = kvs;
+        this.coordinationSequence = coordinationSequence;
+    }
+
+    public static CoordinationStore create(KeyValueService kvs, byte[] coordinationSequence) {
+        CoordinationStore coordinationStore = new KeyValueServiceCoordinationStore(kvs, coordinationSequence);
+        kvs.createTable(AtlasDbConstants.COORDINATION_TABLE, COORDINATION_TABLE_METADATA.persistToBytes());
+        return coordinationStore;
+    }
+
+    @Override
+    public Optional<ByteString> getValue(long sequenceNumber) {
+        Preconditions.checkState(
+                sequenceNumber > 0,
+                "Only positive sequence numbers are supported, but found %s",
+                sequenceNumber);
+        Cell targetCell = getCellForSequence(sequenceNumber);
```

Comment:
Extract a method that takes a sequence number and returns Optional<Value> to get rid of duplication in getCoordinationValue?